base/reduce.jl

```diff
@@ -296,46 +296,95 @@ function extrema(itr)
     return (vmin, vmax)
 end
 
-## all & any
+## any
 
-function mapfoldl(f, ::AndFun, itr)
+any(itr)    = any(IdFun(), itr)
+any(f, itr) = any(specialized_unary(f), itr)
+
+function any(f::IdFun, itr)
     for x in itr
-        !f(x) && return false
+        x && return true
     end
-    return true
+    return false
 end
```

Comment:
Oh yeah, that's right. Although I am not even sure I can use IdFun because `identity` is defined after any is first used (cause of latest build failure). Is there problem in moving `identity`'s  definition earlier?